src/GitHub.Exports.Reactive/Collections/TrackingCollection.cs

```diff
@@ -0,0 +1,842 @@
+        /// collection to be resorted and refiltered.
+        /// </summary>
+        /// <param name="theComparer">The comparer method for sorting, or null if not sorting</param>
+        public void SetComparer(Func<T, T, int> theComparer)
+        {
+            if (disposed)
+                throw new ObjectDisposedException("TrackingCollection");
+            SetAndRecalculateSort(theComparer);
+            SetAndRecalculateFilter(filter);
+        }
+
+        /// <summary>
+        /// Set a new filter. This will cause the collection to be filtered
+        /// </summary>
+        /// <param name="theFilter">The new filter, or null to not have any filtering</param>
+        public void SetFilter(Func<T, int, IList<T>, bool> theFilter)
+        {
+            if (disposed)
+                throw new ObjectDisposedException("TrackingCollection");
+            SetAndRecalculateFilter(theFilter);
+        }
+
+        public IDisposable Subscribe()
+        {
+            if (source == null)
+                throw new InvalidOperationException("No source observable has been set. Call Listen or pass an observable to the constructor");
+            if (disposed)
+                throw new ObjectDisposedException("TrackingCollection");
+            disposables.Add(source.Subscribe());
+            return this;
+        }
+
+        public IDisposable Subscribe(Action<T> onNext, Action onCompleted)
+        {
+            if (source == null)
+                throw new InvalidOperationException("No source observable has been set. Call Listen or pass an observable to the constructor");
+            if (disposed)
+                throw new ObjectDisposedException("TrackingCollection");
+            disposables.Add(source.Subscribe(onNext, onCompleted));
+            return this;
+        }
+
+        public void AddItem(T item)
+        {
+            if (disposed)
+                throw new ObjectDisposedException("TrackingCollection");
+            queue.Enqueue(item);
+        }
+
+        public void RemoveItem(T item)
+        {
+            if (disposed)
+                throw new ObjectDisposedException("TrackingCollection");
+            var position = GetIndexUnfiltered(item);
+            if (position < 0)
+                return;
+            // unfiltered list update
+            original.Remove(item);
+            sortedIndexCache.Remove(item);
+            UpdateIndexCache(original.Count - 1, position, original, sortedIndexCache);
+
+            // filtered list update
+            var index = GetIndexFiltered(item);
+            InternalRemoveItem(item);
+            RecalculateFilter(original, index, position, original.Count);
+        }
+
+        void SetAndRecalculateSort(Func<T, T, int> compare)
+        {
+            comparer = compare;
+            var list = filter != null ? original : Items as List<T>;
+            RecalculateSort(list, 0, list.Count);
```

Comment:
`list` in theory could be null here because `Items` doesn't have to be a `List<T>`. In practice, it always will be in this specific case unless the implementation of `ObservableCollection<T>` or `Collection<T>` changes. So I think we should assert that `list != null` before this line.